src/controllers/movie_controller.go

```diff
@@ -0,0 +1,106 @@
+package controllers
+
+import (
+	"github.com/ELITE-Kinoticketsystem/Backend-KTS/src/.gen/KinoTicketSystem/model"
+	kts_errors "github.com/ELITE-Kinoticketsystem/Backend-KTS/src/errors"
+	"github.com/ELITE-Kinoticketsystem/Backend-KTS/src/models"
+	"github.com/ELITE-Kinoticketsystem/Backend-KTS/src/repositories"
+	"github.com/google/uuid"
+)
+
+type MovieControllerI interface {
+	// Movie
+	GetMovies() (*[]model.Movies, *models.KTSError)
+	GetMovieById(movieId *uuid.UUID) (*model.Movies, *models.KTSError)
+	GetMovieByName(name *string) (*model.Movies, *models.KTSError)
+	CreateMovie(movie *model.Movies) *models.KTSError
+	UpdateMovie(movie *model.Movies) *models.KTSError
+	DeleteMovie(movieId *uuid.UUID) *models.KTSError
+
+	// Combine Movie and Genre
+	// AddMovieGenre(movieId *uuid.UUID, genreId *uuid.UUID) *models.KTSError
+	// RemoveMovieGenre(movieId *uuid.UUID, genreId *uuid.UUID) *models.KTSError
+
+	// One Movie with all Genres
+	GetMovieByIdWithGenre(movieId *uuid.UUID) (*models.MovieWithGenres, *models.KTSError)
+
+	// All Movies with all Genres - Grouped by Movie
+	GetMoviesWithGenres() (*[]models.MovieWithGenres, *models.KTSError)
+
+	GetMovieByIdWithEverything(movieId *uuid.UUID) (*models.MovieWithEverything, *models.KTSError)
+}
+
+type MovieController struct {
+	MovieRepo      repositories.MovieRepositoryI
+	MovieGenreRepo repositories.MovieGenreRepositoryI
+}
+
+// Movie
+func (mc *MovieController) GetMovies() (*[]model.Movies, *models.KTSError) {
+	movies, kts_errors := mc.MovieRepo.GetMovies()
+	if kts_errors != nil {
+		return nil, kts_errors
+	}
+	return movies, nil
+}
+
+func (mc *MovieController) GetMovieById(movieId *uuid.UUID) (*model.Movies, *models.KTSError) {
+	movie, kts_errors := mc.MovieRepo.GetMovieById(movieId)
+	if kts_errors != nil {
+		return nil, kts_errors
+	}
+	return movie, nil
+}
+
+func (mc *MovieController) GetMovieByName(name *string) (*model.Movies, *models.KTSError) {
+	movie, kts_errors := mc.MovieRepo.GetMovieByName(name)
+	if kts_errors != nil {
+		return nil, kts_errors
+	}
+	return movie, nil
+}
+
+func (mc *MovieController) CreateMovie(movie *model.Movies) *models.KTSError {
```

Comment:
Why not implemented?

Reply:
Because it will take some time to develop and the branch is already quite big. Will start doing it next as soon as this is merged. Also i think its more important for the frontend to have the Get-Operations sind the POST Operations are not developed yet in the frontend